mayastor/src/grpc.rs

```diff
@@ -0,0 +1,286 @@
+        locally! { async move {
+            nexus_lookup(&msg.uuid)?.add_child(&msg.uri).await.map(|_| ())
+        }};
+
+        Ok(Response::new(Null {}))
+    }
+
+    async fn remove_child_nexus(
+        &self,
+        request: Request<RemoveChildNexusRequest>,
+    ) -> Result<Response<Null>> {
+        let msg = request.into_inner();
+        locally! { async move {
+            nexus_lookup(&msg.uuid)?.remove_child(&msg.uri).await
+        }};
+
+        Ok(Response::new(Null {}))
+    }
+
+    async fn publish_nexus(
+        &self,
+        request: Request<PublishNexusRequest>,
+    ) -> Result<Response<PublishNexusReply>> {
+        let msg = request.into_inner();
+
+        if msg.key != "" && msg.key.len() != 16 {
+            return Err(nexus_bdev::Error::InvalidKey {}.into());
+        }
+
+        let key: Option<String> = if msg.key == "" {
+            None
+        } else {
+            Some(msg.key.clone())
+        };
+
+        let reply = locally! { async move {
+            nexus_lookup(&msg.uuid)?.share(key).await.map(|device_path| PublishNexusReply { device_path })
+        }};
+
+        Ok(Response::new(reply))
+    }
+
+    async fn unpublish_nexus(
+        &self,
+        request: Request<UnpublishNexusRequest>,
+    ) -> Result<Response<Null>> {
+        locally! { async move {
+            nexus_lookup(&request.into_inner().uuid)?.unshare().await
+        }};
+
+        Ok(Response::new(Null {}))
+    }
+
+    async fn child_operation(
+        &self,
+        request: Request<ChildNexusRequest>,
+    ) -> Result<Response<Null>> {
+        let msg = request.into_inner();
+
+        let onl = match msg.action {
+            1 => Ok(true),
+            0 => Ok(false),
+            _ => Err(Status::invalid_argument("Bad child operation"))
+        }?;
+
+        locally! { async move {
+            let nexus = nexus_lookup(&msg.uuid)?;
+            if onl {
+                nexus.online_child(&msg.uri).await
+            } else {
+                nexus.offline_child(&msg.uri).await
+            }
+        }};
+
+        Ok(Response::new(Null {}))
+    }
+
+    async fn start_rebuild(
+		&self,
+		_request: Request<StartRebuildRequest>,
+	) -> Result<Response<Null>> {
+		todo!()
+	}
+
+	async fn get_rebuild_state(
+		&self,
+		_request: Request<RebuildStateRequest>,
+	) -> Result<Response<RebuildStateReply>> {
+		todo!()
+	}
+
+    async fn get_rebuild_progress(
+		&self,
+		_request: Request<RebuildProgressRequest>,
+	) -> Result<Response<RebuildProgressReply>> {
+		todo!()
+	}
+}
+
+pub async fn grpc_server_init(addr: &str, port: &str)
+        -> std::result::Result<(), ()> {
+    info!("gRPC server configured at address '{}:{}'", addr, port);
+    let saddr = format!("{}:{}", addr, port).parse().unwrap();
+    let svc = Server::builder()
+        .add_service(MayastorServer::new(MayastorGrpc {}))
+        .serve(saddr);
+
+    match svc.await {
+        Ok(_) => Ok(()),
+        Err(_) => Err(()),
```

Comment:
At the every least we should print the error to stdout here?

Reply:
OK, done.

Reply:
Done.

Reply:
Done.